lib/create-testing-library-rule/detect-testing-library-utils.ts

```diff
@@ -614,6 +624,91 @@ export function detectTestingLibraryUtils<
       );
     };
 
+    /**
+     * Determines whether a given node is some reportable `act` util.
+     *
+     * An `act` is reportable if some of these conditions is met:
+     * - it's related to Testing Library module (this depends on Aggressive Reporting)
+     * - it's related to React DOM Test Utils
+     */
+    const isActUtil = (node: TSESTree.Identifier): boolean => {
+      const isTestingLibraryAct = isPotentialTestingLibraryFunction(
+        node,
+        (identifierNodeName, originalNodeName) => {
+          return [identifierNodeName, originalNodeName]
+            .filter(Boolean)
+            .includes('act');
+        }
+      );
+
+      const isReactDomTestUtilsAct = (() => {
+        if (!importedReactDomTestUtilsNode) {
+          return false;
+        }
+        const referenceNode = getReferenceNode(node);
+        const referenceNodeIdentifier = getPropertyIdentifierNode(
+          referenceNode
+        );
+        if (!referenceNodeIdentifier) {
+          return false;
+        }
+
+        const importedUtilSpecifier = findImportSpecifier(
+          node.name,
+          importedReactDomTestUtilsNode
+        );
+        if (!importedUtilSpecifier) {
+          return false;
+        }
+
+        const importDeclaration = (() => {
+          if (isImportDeclaration(importedUtilSpecifier.parent)) {
+            return importedUtilSpecifier.parent;
+          }
+
+          const variableDeclarator = findClosestVariableDeclaratorNode(
+            importedUtilSpecifier
+          );
+
+          if (isCallExpression(variableDeclarator?.init)) {
+            return variableDeclarator?.init;
+          }
+
+          return undefined;
+        })();
+        if (!importDeclaration) {
+          return false;
+        }
+
+        const importDeclarationName = getImportModuleName(importDeclaration);
+        if (!importDeclarationName) {
+          return false;
+        }
+
+        if (importDeclarationName !== REACT_DOM_TEST_UTILS_PACKAGE) {
+          return false;
+        }
+
+        return hasImportMatch(
+          importedUtilSpecifier,
+          referenceNodeIdentifier.name
+        );
```

Comment:
This is actually how `isPotentialTestingLibraryFunction` should check if the import matches the given node.